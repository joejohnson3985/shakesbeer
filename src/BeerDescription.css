.beer-description {
    text-align: center;
    background: #d28b8b;
    position: absolute;
    right: 300px;
    left: 700px;
    top: 300px;
    bottom: 400px;
}

.click-me {
    background: navajowhite;
}