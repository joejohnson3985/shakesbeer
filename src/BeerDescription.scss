@import './variables.scss';

.overlay {
  position: fixed;
  top: 0; 
  left: 0; 
  width: 100%; 
  height: 100%;
  // display: flex;
  // justify-content: center;
  // align-items: center;
  background-color: rgba(90, 90, 90, 0.5);
  cursor: pointer;
  z-index: 10;
}

.beer-description {
  position: absolute;
  margin: -150px 0 0 -300px;
  top: 50%;
  left: 50%;
  border-radius: 3px;
  width: 600px;
  background: #fff;
  z-index:11;
  border: 2px solid $secondary-color;
  cursor: default;
  
  h3 {
    text-align: center;
    color: $primary-color;
  }
}

.beer-stuff {
  margin-left: 40px;
  color: $third-color;
  font-size: 20px;
}

.click-me {
  color: #fff;
  font-size: 18px;
  margin: 20px 50px;
  width: 500px;
  height: 35px;
  transition: background-color .3s ease-in-out;
  cursor: pointe;

  background-color: $secondary-color;

  &:hover {
  background-color: $primary-color;
  }
}