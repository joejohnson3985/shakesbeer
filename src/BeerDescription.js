import React, { Component } from 'react';
import './BeerDescription.css'

class BeerDescription extends Component {
    render() {
        return (
            <div className='beer-description'>
                <h3>{this.props.selectedBeer.name}</h3>
                <div className='beer-stuff'>
                    <p>Style: {this.props.selectedBeer.style}</p>
                    <p>IBU: {this.props.selectedBeer.ibu}</p>
                    <p>ABV: {this.props.selectedBeer.abv}</p>
                </div>
                <p className='click-me'
                   role='button'
                   onClick={this.props.toggle}>Click ME!</p>
            </div>
        );
    }
}

export default BeerDescription;