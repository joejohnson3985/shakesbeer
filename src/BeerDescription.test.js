import React from 'react';
import BeerDescription from './BeerDescription.js';
import { shallow } from 'enzyme';

const mockData =
  {
    "FIELD1": 44,
    "abv": 0.055,
    "ibu": null,
    "id": 2324,
    "name": "Cast Iron Oatmeal Brown",
    "style": "American Brown Ale",
    "brewery_id": 60,
    "ounces": 12
  };

const mockFunc = jest.fn()

describe('BeerDescription', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow (
            <BeerDescription selectedBeer={mockData}
                             toggle={mockFunc}
            /> 
        )
    });

    it('Should match a snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    });

    it('Should simulate a click', () => {
        wrapper.find('.click-me').simulate('click');
        expect(mockFunc).toHaveBeenCalled();
    });
});