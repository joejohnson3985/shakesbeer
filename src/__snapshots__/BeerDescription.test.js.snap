// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeerDescription Should match a snapshot 1`] = `
<div
  className="beer-description"
>
  <h3>
    Cast Iron Oatmeal Brown
  </h3>
  <div
    className="beer-stuff"
  >
    <p>
      Style: 
      American Brown Ale
    </p>
    <p>
      IBU: 
    </p>
    <p>
      ABV: 
      0.055
    </p>
  </div>
  <p
    className="click-me"
    onClick={[MockFunction]}
    role="button"
  >
    Click ME!
  </p>
</div>
`;
